internal/ui/exec_process/exec_process.go

```diff
@@ -0,0 +1,127 @@
+package exec_process
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/idursun/jjui/internal/jj"
+	"github.com/idursun/jjui/internal/ui/common"
+	"github.com/idursun/jjui/internal/ui/context"
+)
+
+func ExecMsgFromLine(prompt string, line string) common.ExecMsg {
+	line = strings.TrimSpace(line)
+	switch prompt {
+	case common.ExecShell.Prompt:
+		return common.ExecMsg{
+			Line: line,
+			Mode: common.ExecShell,
+		}
+	default:
+		return common.ExecMsg{
+			Line: line,
+			Mode: common.ExecJJ,
+		}
+	}
+}
+
+func ExecLine(ctx *context.MainContext, msg common.ExecMsg) tea.Cmd {
+	replacements := ctx.CreateReplacements()
+	args := splitCommandArgs(msg.Line)
+	args = jj.TemplatedArgs(args, replacements)
+	switch msg.Mode {
+	case common.ExecJJ:
+		return exec_program("jj", args)
+	case common.ExecShell:
+		// user input is run via `$SHELL -c`` to support user specifying command lines
+		// that have pipes (to a pager) or redirection.
+		program := os.Getenv("SHELL")
+		if len(program) == 0 {
+			program = "sh"
+		}
+		args = []string{
+			"-c",
+			strings.Join(args, " "),
+		}
+		return exec_program(program, args)
+	}
+	return nil
+}
+
+// This is different from command_runner.RunInteractiveCommand.
+// This function does not capture any IO. We want all IO to be given to the program.
+// Upon program termination we ask the user to press a key to return to JJUI.
+// This is useful for programs that would otherwise terminate quickly and just flash.
+func exec_program(program string, args []string) tea.Cmd {
+	p := &process{program: program, args: args}
+	return tea.Batch(
+		common.CommandRunning(args),
```

Comment:
@vic:  CommandRunning should take `program` not only args as we had in the previous patch so that the status line reflect the correct command being run. Otherwise user will see `jj -c whoami` if running `$ whoami`.